# Asthma kasalligi bo'yicha kasallanganlar dataseti

### Pandasni import qilib chaqirib olish va pd.read_csv funksiyasidan foydalanib kerakli bo'lgan csv faylni ochib olish. Bu funksiya data frame qaytaradi!

In [2]:
import pandas as pd

df = pd.read_csv('asthma_disease_data.csv')
df

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0          5034   63       0          1               0  15.848744        0   
1          5035   26       1          2               2  22.757042        0   
2          5036   57       0          2               1  18.395396        0   
3          5037   40       1          2               1  38.515278        0   
4          5038   61       0          0               3  19.283802        0   
...         ...  ...     ...        ...             ...        ...      ...   
2387       7421   43       1          0               2  29.059613        0   
2388       7422   18       1          0               1  20.740850        0   
2389       7423   54       0          3               2  37.079560        0   
2390       7424   46       1          0               2  23.444712        0   
2391       7425   26       1          0               0  28.123021        1   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
0             0.894448     5.488696      8.701003  ...          1.369051   
1             5.897329     6.341014      5.153966  ...          2.197767   
2             6.739367     9.196237      6.840647  ...          1.698011   
3             1.404503     5.826532      4.253036  ...          3.032037   
4             4.604493     3.127048      9.625799  ...          3.470589   
...                ...          ...           ...  ...               ...   
2387          3.019854     6.119637      8.300960  ...          3.125249   
2388          5.805180     4.386992      7.731192  ...          1.132977   
2389          4.735169     8.214064      7.483521  ...          1.685962   
2390          9.672637     7.362861      6.717272  ...          3.481549   
2391          1.613138     7.412878      8.512253  ...          2.280613   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
0            4.941206         0                  0               1         0   
1            1.702393         1                  0               0         1   
2            5.022553         1                  1               1         0   
3            2.300159         1                  0               1         1   
4            3.067944         1                  1               1         0   
...               ...       ...                ...             ...       ...   
2387         5.166032         0                  1               0         0   
2388         5.509502         0                  0               0         1   
2389         3.346877         1                  0               1         1   
2390         1.713274         0                  1               1         0   
2391         2.453284         1                  1               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
0                     0                1          0       Dr_Confid  
1                     1                1          0       Dr_Confid  
2                     1                1          0       Dr_Confid  
3                     1                0          0       Dr_Confid  
4                     0                1          0       Dr_Confid  
...                 ...              ...        ...             ...  
2387                  0                1          1       Dr_Confid  
2388                  1                0          1       Dr_Confid  
2389                  0                1          1       Dr_Confid  
2390                  1                1          0       Dr_Confid  
2391                  1                0          0       Dr_Confid  

[2392 rows x 29 columns]

Patient ID: from 5034 to 7425, Bemorlarni ID si
Age: from 5 to 80 years
Gender: 0 - Male, 1 - Female
EthnicityL: 0: Caucasian
            1: African American
            2: Asian
            3: Other
Education level: 0: None
                 1: High School
                 2: Bachelor's
                 3: Higher
BMI: Body mass index. from 15 to 40 (18.5 to 24.9 - healthy, 25.0 to 29.9 - overweight, 30.0 or higher - obese)
Smoking: 1 - Yes, 0 - No
PhysicalActivity: Weekly physical activity in hours, ranging from 0 to 10.
DietQuality: Diet quality score, ranging from 0 to 10.
SleepQuality: Sleep quality score, ranging from 4 to 10.

I worked with this indicators!

### df.head() funksiyasi ichiga qiymat berilmagan taqdirda faqat jadvalning dastlabki 5 ta qatorini chiqaradi. Agar ichiga raqam berilsa osha son miqdoricha qatorlarni chiqaradi.

In [3]:
df.head()

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0       5034   63       0          1               0  15.848744        0   
1       5035   26       1          2               2  22.757042        0   
2       5036   57       0          2               1  18.395396        0   
3       5037   40       1          2               1  38.515278        0   
4       5038   61       0          0               3  19.283802        0   

   PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
0          0.894448     5.488696      8.701003  ...          1.369051   
1          5.897329     6.341014      5.153966  ...          2.197767   
2          6.739367     9.196237      6.840647  ...          1.698011   
3          1.404503     5.826532      4.253036  ...          3.032037   
4          4.604493     3.127048      9.625799  ...          3.470589   

   LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
0         4.941206         0                  0               1         0   
1         1.702393         1                  0               0         1   
2         5.022553         1                  1               1         0   
3         2.300159         1                  0               1         1   
4         3.067944         1                  1               1         0   

   NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
0                  0                1          0       Dr_Confid  
1                  1                1          0       Dr_Confid  
2                  1                1          0       Dr_Confid  
3                  1                0          0       Dr_Confid  
4                  0                1          0       Dr_Confid  

[5 rows x 29 columns]

### df.info() - methodi dT frame haqida ma'lumot beradi. Asosiy ma'lumotlar Biror qator null mi null emasmi shuni korib olish mumkin. umumiy nechta column nechta qator borligini va har bir columnlarning data typeni bilib olishimiz mumkin. 

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2392 entries, 0 to 2391
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   PatientID               2392 non-null   int64  
 1   Age                     2392 non-null   int64  
 2   Gender                  2392 non-null   int64  
 3   Ethnicity               2392 non-null   int64  
 4   EducationLevel          2392 non-null   int64  
 5   BMI                     2392 non-null   float64
 6   Smoking                 2392 non-null   int64  
 7   PhysicalActivity        2392 non-null   float64
 8   DietQuality             2392 non-null   float64
 9   SleepQuality            2392 non-null   float64
 10  PollutionExposure       2392 non-null   float64
 11  PollenExposure          2392 non-null   float64
 12  DustExposure            2392 non-null   float64
 13  PetAllergy              2392 non-null   int64  
 14  FamilyHistoryAsthma     2392 non-null   

### Pandasda 2 ta katta obyekt bor: 1 - Data frame, 2 - Series! Series bu biz data frame dan cma'lum bir columnni ajratib olganimizda hosil bo'ladi. Bunda bitta o'zgaruvchi ochilib age ustuni shu o'zgaruvchiga tenglanyapti typpe: series

In [9]:
age = df['Age']
type(age)
age

0       63
1       26
2       57
3       40
4       61
        ..
2387    43
2388    18
2389    54
2390    46
2391    26
Name: Age, Length: 2392, dtype: int64

### Bundan tashqqari ustunni nuqta bilan ham ajratib olish mumkin. lekin bu tavsiya etilmaydi chunki chalg'itadi!

In [10]:
age = df.Age
age

0       63
1       26
2       57
3       40
4       61
        ..
2387    43
2388    18
2389    54
2390    46
2391    26
Name: Age, Length: 2392, dtype: int64

### Biz bundan tashqari ustunlar ustida amallar qilsak bo'ladi (<, >, <=, >=, ==, !=). Bu amallar bizga boolean(True or False) qaytaradi.

In [15]:
age > 30

0        True
1       False
2        True
3        True
4        True
        ...  
2387     True
2388    False
2389     True
2390     True
2391    False
Name: Age, Length: 2392, dtype: bool

### Quyidagi amal faqat 30 yoshdan yuqorilarni olib beradi. Type: data frame

In [16]:
df[age > 30]

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0          5034   63       0          1               0  15.848744        0   
2          5036   57       0          2               1  18.395396        0   
3          5037   40       1          2               1  38.515278        0   
4          5038   61       0          0               3  19.283802        0   
6          5040   45       1          1               1  30.245954        1   
...         ...  ...     ...        ...             ...        ...      ...   
2383       7417   62       1          0               2  20.033593        0   
2384       7418   31       0          0               2  31.821008        1   
2387       7421   43       1          0               2  29.059613        0   
2389       7423   54       0          3               2  37.079560        0   
2390       7424   46       1          0               2  23.444712        0   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
0             0.894448     5.488696      8.701003  ...          1.369051   
2             6.739367     9.196237      6.840647  ...          1.698011   
3             1.404503     5.826532      4.253036  ...          3.032037   
4             4.604493     3.127048      9.625799  ...          3.470589   
6             9.371784     7.030507      5.746128  ...          2.995100   
...                ...          ...           ...  ...               ...   
2383          5.276410     9.545336      6.279582  ...          1.827517   
2384          8.516835     3.532328      9.442670  ...          1.883568   
2387          3.019854     6.119637      8.300960  ...          3.125249   
2389          4.735169     8.214064      7.483521  ...          1.685962   
2390          9.672637     7.362861      6.717272  ...          3.481549   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
0            4.941206         0                  0               1         0   
2            5.022553         1                  1               1         0   
3            2.300159         1                  0               1         1   
4            3.067944         1                  1               1         0   
6            1.701512         1                  1               1         0   
...               ...       ...                ...             ...       ...   
2383         3.757340         1                  0               1         1   
2384         2.592148         0                  0               0         0   
2387         5.166032         0                  1               0         0   
2389         3.346877         1                  0               1         1   
2390         1.713274         0                  1               1         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
0                     0                1          0       Dr_Confid  
2                     1                1          0       Dr_Confid  
3                     1                0          0       Dr_Confid  
4                     0                1          0       Dr_Confid  
6                     0                0          0       Dr_Confid  
...                 ...              ...        ...             ...  
2383                  1                0          0       Dr_Confid  
2384                  1                1          1       Dr_Confid  
2387                  0                1          1       Dr_Confid  
2389                  0                1          1       Dr_Confid  
2390                  1                1          0       Dr_Confid  

[1573 rows x 29 columns]

### Quyidagi holat Etnik jihatdan faqat Osiyoda uchraydigan bemorlarni chiqarib beradi!

In [18]:
df[df['Ethnicity'] == 2]

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
1          5035   26       1          2               2  22.757042        0   
2          5036   57       0          2               1  18.395396        0   
3          5037   40       1          2               1  38.515278        0   
5          5039   21       0          2               0  21.812975        0   
20         5054   40       1          2               1  23.588420        1   
...         ...  ...     ...        ...             ...        ...      ...   
2344       7378   26       1          2               2  17.141151        0   
2346       7380   37       1          2               1  39.335351        0   
2356       7390    6       0          2               1  36.501742        0   
2362       7396   56       1          2               2  38.132535        0   
2366       7400   35       0          2               0  22.154735        0   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
1             5.897329     6.341014      5.153966  ...          2.197767   
2             6.739367     9.196237      6.840647  ...          1.698011   
3             1.404503     5.826532      4.253036  ...          3.032037   
5             0.470044     1.759118      9.549262  ...          2.328191   
20            9.024920     0.911513      9.913215  ...          2.420994   
...                ...          ...           ...  ...               ...   
2344          8.852679     7.575876      5.366718  ...          1.159288   
2346          1.019512     6.763305      4.804355  ...          3.441937   
2356          2.020731     4.723211      8.385088  ...          3.252228   
2362          4.588728     9.243218      7.987792  ...          3.961105   
2366          0.032589     0.067117      9.038348  ...          2.145319   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
1            1.702393         1                  0               0         1   
2            5.022553         1                  1               1         0   
3            2.300159         1                  0               1         1   
5            5.898515         1                  0               1         0   
20           3.443648         1                  1               0         1   
...               ...       ...                ...             ...       ...   
2344         4.841812         0                  0               0         0   
2346         3.566303         0                  0               1         0   
2356         4.620786         1                  0               0         0   
2362         5.523770         1                  1               0         1   
2366         5.908935         1                  0               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
1                     1                1          0       Dr_Confid  
2                     1                1          0       Dr_Confid  
3                     1                0          0       Dr_Confid  
5                     0                1          0       Dr_Confid  
20                    1                0          0       Dr_Confid  
...                 ...              ...        ...             ...  
2344                  0                0          0       Dr_Confid  
2346                  0                1          1       Dr_Confid  
2356                  0                0          0       Dr_Confid  
2362                  0                1          1       Dr_Confid  
2366                  1                1          1       Dr_Confid  

[229 rows x 29 columns]

### df.shape - atribute hisoblanib data frameda nechta qator va nechta ustun borligini tuplega solib qaytaradi.

In [21]:
shape = df.shape
shape

(2392, 29)

### df.describe() - method hisoblanib data frameni umumiy holda tasvirlaydi. Bunda min, max, std, count, quartile(25%, 50%, 75%) bo'ladi

In [22]:
df.describe()

PatientID          Age       Gender    Ethnicity  EducationLevel  \
count  2392.000000  2392.000000  2392.000000  2392.000000     2392.000000   
mean   6229.500000    42.137960     0.493311     0.669732        1.307274   
std     690.655244    21.606655     0.500060     0.986120        0.898242   
min    5034.000000     5.000000     0.000000     0.000000        0.000000   
25%    5631.750000    23.000000     0.000000     0.000000        1.000000   
50%    6229.500000    42.000000     0.000000     0.000000        1.000000   
75%    6827.250000    61.000000     1.000000     1.000000        2.000000   
max    7425.000000    79.000000     1.000000     3.000000        3.000000   

               BMI      Smoking  PhysicalActivity  DietQuality  SleepQuality  \
count  2392.000000  2392.000000       2392.000000  2392.000000   2392.000000   
mean     27.244877     0.141722          5.051786     5.022867      7.019012   
std       7.201628     0.348838          2.903574     2.909980      1.732475   
min      15.031803     0.000000          0.001740     0.003031      4.001437   
25%      20.968313     0.000000          2.578333     2.432043      5.498500   
50%      27.052202     0.000000          5.016881     5.115383      6.975839   
75%      33.555903     0.000000          7.540234     7.544216      8.526950   
max      39.985611     1.000000          9.995809     9.999904      9.996235   

       ...  GastroesophagealReflux  LungFunctionFEV1  LungFunctionFVC  \
count  ...             2392.000000       2392.000000      2392.000000   
mean   ...                0.158027          2.548564         3.741270   
std    ...                0.364842          0.861809         1.303689   
min    ...                0.000000          1.000459         1.500045   
25%    ...                0.000000          1.824113         2.607489   
50%    ...                0.000000          2.553244         3.734982   
75%    ...                0.000000          3.292897         4.864121   
max    ...                1.000000          3.999719         5.999421   

          Wheezing  ShortnessOfBreath  ChestTightness     Coughing  \
count  2392.000000        2392.000000     2392.000000  2392.000000   
mean      0.596154           0.500418        0.503344     0.503344   
std       0.490770           0.500104        0.500093     0.500093   
min       0.000000           0.000000        0.000000     0.000000   
25%       0.000000           0.000000        0.000000     0.000000   
50%       1.000000           1.000000        1.000000     1.000000   
75%       1.000000           1.000000        1.000000     1.000000   
max       1.000000           1.000000        1.000000     1.000000   

       NighttimeSymptoms  ExerciseInduced    Diagnosis  
count        2392.000000      2392.000000  2392.000000  
mean            0.602425         0.604933     0.051839  
std             0.489499         0.488967     0.221749  
min             0.000000         0.000000     0.000000  
25%             0.000000         0.000000     0.000000  
50%             1.000000         1.000000     0.000000  
75%             1.000000         1.000000     0.000000  
max             1.000000         1.000000     1.000000  

[8 rows x 28 columns]

### df.values - atribute hisoblanib data frame ichidagi hamma qatorlarni list ichida list qilib solib beradi

In [23]:
df.values

array([[5034, 63, 0, ..., 1, 0, 'Dr_Confid'],
       [5035, 26, 1, ..., 1, 0, 'Dr_Confid'],
       [5036, 57, 0, ..., 1, 0, 'Dr_Confid'],
       ...,
       [7423, 54, 0, ..., 1, 1, 'Dr_Confid'],
       [7424, 46, 1, ..., 1, 0, 'Dr_Confid'],
       [7425, 26, 1, ..., 0, 0, 'Dr_Confid']], dtype=object)

### df.columns columns nomlarini qaytaradi. df.columns.tolist esa column nomlarini listga solib beradi

In [24]:
df.columns

Index(['PatientID', 'Age', 'Gender', 'Ethnicity', 'EducationLevel', 'BMI',
       'Smoking', 'PhysicalActivity', 'DietQuality', 'SleepQuality',
       'PollutionExposure', 'PollenExposure', 'DustExposure', 'PetAllergy',
       'FamilyHistoryAsthma', 'HistoryOfAllergies', 'Eczema', 'HayFever',
       'GastroesophagealReflux', 'LungFunctionFEV1', 'LungFunctionFVC',
       'Wheezing', 'ShortnessOfBreath', 'ChestTightness', 'Coughing',
       'NighttimeSymptoms', 'ExerciseInduced', 'Diagnosis', 'DoctorInCharge'],
      dtype='object')

In [25]:
df.columns.tolist()

['PatientID',
 'Age',
 'Gender',
 'Ethnicity',
 'EducationLevel',
 'BMI',
 'Smoking',
 'PhysicalActivity',
 'DietQuality',
 'SleepQuality',
 'PollutionExposure',
 'PollenExposure',
 'DustExposure',
 'PetAllergy',
 'FamilyHistoryAsthma',
 'HistoryOfAllergies',
 'Eczema',
 'HayFever',
 'GastroesophagealReflux',
 'LungFunctionFEV1',
 'LungFunctionFVC',
 'Wheezing',
 'ShortnessOfBreath',
 'ChestTightness',
 'Coughing',
 'NighttimeSymptoms',
 'ExerciseInduced',
 'Diagnosis',
 'DoctorInCharge']

### Sorting - sortlab beruvchi methodlari ham bor

In [27]:
df.sort_values("Age")

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
703        5737    5       0          0               0  30.163020        0   
2211       7245    5       0          0               1  38.155312        1   
2207       7241    5       1          0               2  37.839054        0   
814        5848    5       0          0               1  17.572297        0   
2081       7115    5       0          1               3  23.508565        0   
...         ...  ...     ...        ...             ...        ...      ...   
68         5102   79       0          0               3  17.786832        0   
1984       7018   79       1          3               0  17.867079        0   
1777       6811   79       0          0               2  28.439224        1   
746        5780   79       1          0               1  32.724673        0   
1641       6675   79       0          0               2  30.030769        0   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
703           2.048262     0.123081      4.619742  ...          1.625130   
2211          5.494052     5.390795      7.125330  ...          2.297852   
2207          9.416716     5.584356      9.207865  ...          2.811539   
814           3.750060     7.641404      8.642893  ...          3.106335   
2081          7.574201     0.331971      9.978663  ...          2.408726   
...                ...          ...           ...  ...               ...   
68            1.933638     0.403664      4.935988  ...          1.197318   
1984          5.380631     0.421013      9.886130  ...          2.201482   
1777          5.271687     5.828416      9.248739  ...          1.050262   
746           5.692000     9.184941      9.394542  ...          1.993555   
1641          9.507565     9.315187      6.127767  ...          1.550747   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
703          4.179803         1                  0               1         0   
2211         3.697025         1                  0               0         0   
2207         1.810222         1                  0               0         0   
814          3.422943         1                  1               0         1   
2081         4.654947         1                  0               1         0   
...               ...       ...                ...             ...       ...   
68           5.610579         0                  0               0         0   
1984         5.299593         0                  0               0         0   
1777         4.621472         0                  1               0         1   
746          5.510775         1                  0               0         1   
1641         3.716878         0                  1               1         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
703                   1                1          0       Dr_Confid  
2211                  0                1          1       Dr_Confid  
2207                  0                1          0       Dr_Confid  
814                   1                1          0       Dr_Confid  
2081                  0                1          0       Dr_Confid  
...                 ...              ...        ...             ...  
68                    0                0          0       Dr_Confid  
1984                  0                1          0       Dr_Confid  
1777                  0                1          0       Dr_Confid  
746                   1                0          0       Dr_Confid  
1641                  1                0          0       Dr_Confid  

[2392 rows x 29 columns]

In [28]:
df.sort_values("Age", ascending=False)

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
1251       6285   79       1          0               1  29.993441        1   
226        5260   79       0          3               1  18.372529        0   
697        5731   79       1          1               1  31.282971        0   
1031       6065   79       1          0               0  18.307355        0   
843        5877   79       0          1               1  34.770417        0   
...         ...  ...     ...        ...             ...        ...      ...   
619        5653    5       1          0               2  25.455246        0   
2348       7382    5       0          1               1  20.936782        0   
1008       6042    5       1          0               1  17.190758        1   
679        5713    5       0          2               1  26.850546        0   
405        5439    5       0          3               2  21.866782        0   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
1251          2.812983     2.213355      9.170472  ...          3.650503   
226           5.427240     2.334628      7.620781  ...          2.272436   
697           7.986946     8.649852      8.293709  ...          1.295325   
1031          9.700644     0.042102      7.876771  ...          1.813745   
843           5.755839     7.581654      5.284070  ...          2.527161   
...                ...          ...           ...  ...               ...   
619           7.003770     3.139850      5.745023  ...          2.325320   
2348          5.277747     2.910634      9.553447  ...          2.434421   
1008          8.524377     8.317265      8.270775  ...          1.903457   
679           6.683361     3.252095      9.050079  ...          1.607371   
405           1.125857     4.762594      6.870321  ...          1.004808   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
1251         5.604767         0                  1               0         1   
226          3.338476         0                  1               1         0   
697          5.939451         1                  1               1         0   
1031         3.206849         1                  1               0         0   
843          3.342889         1                  1               1         0   
...               ...       ...                ...             ...       ...   
619          4.643632         0                  0               0         0   
2348         5.941801         0                  0               1         1   
1008         2.992458         0                  1               0         0   
679          2.755117         0                  1               0         0   
405          3.550239         0                  0               1         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
1251                  1                0          0       Dr_Confid  
226                   1                0          0       Dr_Confid  
697                   0                0          0       Dr_Confid  
1031                  0                1          0       Dr_Confid  
843                   1                1          0       Dr_Confid  
...                 ...              ...        ...             ...  
619                   0                1          0       Dr_Confid  
2348                  1                0          1       Dr_Confid  
1008                  0                0          0       Dr_Confid  
679                   0                0          0       Dr_Confid  
405                   1                1          0       Dr_Confid  

[2392 rows x 29 columns]

In [30]:
df.sort_values(["Age", "BMI"])

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
1008       6042    5       1          0               1  17.190758        1   
814        5848    5       0          0               1  17.572297        0   
1438       6472    5       0          0               1  17.651456        0   
1579       6613    5       0          1               1  18.601088        1   
1784       6818    5       1          0               0  19.462619        1   
...         ...  ...     ...        ...             ...        ...      ...   
71         5105   79       1          1               2  38.421666        0   
1933       6967   79       0          0               0  38.578388        1   
1611       6645   79       0          1               2  38.712137        1   
865        5899   79       1          0               3  39.010001        0   
1376       6410   79       0          0               0  39.666881        0   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
1008          8.524377     8.317265      8.270775  ...          1.903457   
814           3.750060     7.641404      8.642893  ...          3.106335   
1438          4.372488     0.794238      4.902879  ...          3.985125   
1579          2.182973     3.723267      5.681424  ...          3.807622   
1784          4.161883     8.511220      9.147175  ...          1.566862   
...                ...          ...           ...  ...               ...   
71            0.696851     2.452365      5.867393  ...          1.609002   
1933          3.261688     6.362565      8.951902  ...          1.928822   
1611          5.689947     1.170514      5.303048  ...          3.291413   
865           8.954119     9.823385      6.626818  ...          1.487352   
1376          9.673421     6.641485      7.342028  ...          3.398467   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
1008         2.992458         0                  1               0         0   
814          3.422943         1                  1               0         1   
1438         2.657408         0                  0               0         1   
1579         1.509617         0                  0               0         0   
1784         3.957622         1                  0               1         0   
...               ...       ...                ...             ...       ...   
71           3.621563         1                  1               0         0   
1933         1.544718         0                  0               1         0   
1611         2.919717         1                  1               0         0   
865          4.254119         1                  0               1         0   
1376         3.993066         1                  0               1         1   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
1008                  0                0          0       Dr_Confid  
814                   1                1          0       Dr_Confid  
1438                  0                1          0       Dr_Confid  
1579                  0                1          0       Dr_Confid  
1784                  0                0          0       Dr_Confid  
...                 ...              ...        ...             ...  
71                    1                1          0       Dr_Confid  
1933                  1                0          0       Dr_Confid  
1611                  1                0          0       Dr_Confid  
865                   1                1          0       Dr_Confid  
1376                  0                1          0       Dr_Confid  

[2392 rows x 29 columns]

Avval age bo'yicha so'rtlab Agar bir hil agelilar bo'lsa keyin Body index bo'yicha teskari tartibda so'rtlayapti

In [31]:
df.sort_values(["Age", "BMI"], ascending=[True, False])

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
2211       7245    5       0          0               1  38.155312        1   
2207       7241    5       1          0               2  37.839054        0   
990        6024    5       1          3               1  36.251830        0   
1826       6860    5       0          1               0  36.172342        0   
2377       7411    5       1          1               2  35.479080        0   
...         ...  ...     ...        ...             ...        ...      ...   
226        5260   79       0          3               1  18.372529        0   
1031       6065   79       1          0               0  18.307355        0   
1984       7018   79       1          3               0  17.867079        0   
68         5102   79       0          0               3  17.786832        0   
784        5818   79       1          3               2  15.325347        0   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
2211          5.494052     5.390795      7.125330  ...          2.297852   
2207          9.416716     5.584356      9.207865  ...          2.811539   
990           8.847617     3.270534      7.442324  ...          3.020129   
1826          5.417269     4.478271      5.777935  ...          3.277718   
2377          4.571440     4.228623      7.428542  ...          2.675931   
...                ...          ...           ...  ...               ...   
226           5.427240     2.334628      7.620781  ...          2.272436   
1031          9.700644     0.042102      7.876771  ...          1.813745   
1984          5.380631     0.421013      9.886130  ...          2.201482   
68            1.933638     0.403664      4.935988  ...          1.197318   
784           4.642249     8.049932      7.565453  ...          2.590358   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
2211         3.697025         1                  0               0         0   
2207         1.810222         1                  0               0         0   
990          2.523197         1                  1               0         0   
1826         3.140847         1                  0               1         1   
2377         1.737707         0                  1               0         1   
...               ...       ...                ...             ...       ...   
226          3.338476         0                  1               1         0   
1031         3.206849         1                  1               0         0   
1984         5.299593         0                  0               0         0   
68           5.610579         0                  0               0         0   
784          4.909318         0                  0               0         1   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
2211                  0                1          1       Dr_Confid  
2207                  0                1          0       Dr_Confid  
990                   1                1          0       Dr_Confid  
1826                  1                0          0       Dr_Confid  
2377                  0                1          1       Dr_Confid  
...                 ...              ...        ...             ...  
226                   1                0          0       Dr_Confid  
1031                  0                1          0       Dr_Confid  
1984                  0                1          0       Dr_Confid  
68                    0                0          0       Dr_Confid  
784                   1                0          0       Dr_Confid  

[2392 rows x 29 columns]

### Bir nechta columnlarni chiqarish

In [32]:
df[['Age', 'Gender', 'Ethnicity']]

Age  Gender  Ethnicity
0      63       0          1
1      26       1          2
2      57       0          2
3      40       1          2
4      61       0          0
...   ...     ...        ...
2387   43       1          0
2388   18       1          0
2389   54       0          3
2390   46       1          0
2391   26       1          0

[2392 rows x 3 columns]

Bir nechta columnlarni o'zgaruvchiga tenglab qo'yish

In [33]:
columns = ['Age', 'Gender', 'Ethnicity']
df[columns]

Age  Gender  Ethnicity
0      63       0          1
1      26       1          2
2      57       0          2
3      40       1          2
4      61       0          0
...   ...     ...        ...
2387   43       1          0
2388   18       1          0
2389   54       0          3
2390   46       1          0
2391   26       1          0

[2392 rows x 3 columns]

Columns ustida turli holatlar

In [34]:
is_male = df['Gender'] == 0
is_asia = df['Ethnicity'] == 2

df[is_male & is_asia]

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
2          5036   57       0          2               1  18.395396        0   
5          5039   21       0          2               0  21.812975        0   
21         5055   72       0          2               1  33.550266        0   
22         5056   53       0          2               1  37.923024        0   
40         5074    7       0          2               2  16.343191        0   
...         ...  ...     ...        ...             ...        ...      ...   
2251       7285   27       0          2               1  33.115695        0   
2319       7353   26       0          2               2  25.396059        0   
2334       7368   75       0          2               2  30.437967        0   
2356       7390    6       0          2               1  36.501742        0   
2366       7400   35       0          2               0  22.154735        0   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
2             6.739367     9.196237      6.840647  ...          1.698011   
5             0.470044     1.759118      9.549262  ...          2.328191   
21            1.514889     6.790537      4.084064  ...          3.504798   
22            3.741621     2.571249      7.854012  ...          3.569894   
40            0.931879     5.863282      5.431136  ...          2.027385   
...                ...          ...           ...  ...               ...   
2251          3.647065     0.218472      6.506372  ...          3.476001   
2319          7.752296     6.350328      4.993477  ...          3.064922   
2334          9.434656     7.123279      9.530553  ...          3.317028   
2356          2.020731     4.723211      8.385088  ...          3.252228   
2366          0.032589     0.067117      9.038348  ...          2.145319   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
2            5.022553         1                  1               1         0   
5            5.898515         1                  0               1         0   
21           2.104287         0                  0               1         0   
22           2.756690         0                  0               1         0   
40           3.982930         0                  0               1         0   
...               ...       ...                ...             ...       ...   
2251         2.760895         1                  0               0         1   
2319         4.341987         1                  1               0         0   
2334         2.231574         0                  0               0         1   
2356         4.620786         1                  0               0         0   
2366         5.908935         1                  0               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
2                     1                1          0       Dr_Confid  
5                     0                1          0       Dr_Confid  
21                    0                0          0       Dr_Confid  
22                    0                0          0       Dr_Confid  
40                    0                1          0       Dr_Confid  
...                 ...              ...        ...             ...  
2251                  1                1          0       Dr_Confid  
2319                  1                1          0       Dr_Confid  
2334                  0                1          0       Dr_Confid  
2356                  0                0          0       Dr_Confid  
2366                  1                1          1       Dr_Confid  

[120 rows x 29 columns]

.isin() - method kelayotgan yosh tganlangan yoshlar ichida bo'lsa chiqaradi

In [36]:
is_58_or_59 = df['Age'].isin([58, 59])
df[is_58_or_59]

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
56         5090   58       0          0               1  15.641879        0   
78         5112   58       1          0               1  18.503524        0   
126        5160   58       0          0               3  39.101842        0   
234        5268   58       0          0               2  23.356623        0   
358        5392   58       0          0               1  34.636258        0   
...         ...  ...     ...        ...             ...        ...      ...   
2237       7271   58       0          1               1  26.459200        0   
2327       7361   59       0          1               1  36.136556        0   
2339       7373   59       1          0               2  35.551043        0   
2360       7394   58       1          0               3  28.265492        0   
2368       7402   58       0          0               1  33.382904        0   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
56            2.939381     9.881885      9.437460  ...          3.879503   
78            6.871905     5.725532      8.736870  ...          1.056951   
126           0.234806     8.946924      4.991436  ...          1.470990   
234           4.890278     2.294554      9.676931  ...          1.417626   
358           9.506625     9.965655      5.905376  ...          1.497544   
...                ...          ...           ...  ...               ...   
2237          7.630881     1.771875      4.238033  ...          3.918281   
2327          9.538894     7.038338      9.210262  ...          3.311682   
2339          9.484381     6.790636      7.547532  ...          2.025401   
2360          9.773646     4.937273      7.376810  ...          3.707605   
2368          2.017431     6.735866      4.008474  ...          2.822054   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
56           2.245368         1                  1               1         1   
78           3.544918         1                  1               1         0   
126          4.235386         0                  1               0         0   
234          5.526718         0                  1               1         1   
358          2.372843         1                  0               1         0   
...               ...       ...                ...             ...       ...   
2237         1.696235         0                  1               1         1   
2327         1.743094         0                  0               0         1   
2339         5.878464         1                  0               1         1   
2360         3.849071         1                  0               0         0   
2368         5.799194         0                  1               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
56                    1                0          0       Dr_Confid  
78                    0                0          0       Dr_Confid  
126                   1                1          0       Dr_Confid  
234                   1                0          0       Dr_Confid  
358                   1                0          0       Dr_Confid  
...                 ...              ...        ...             ...  
2237                  1                1          1       Dr_Confid  
2327                  1                1          0       Dr_Confid  
2339                  1                0          0       Dr_Confid  
2360                  0                1          1       Dr_Confid  
2368                  0                1          0       Dr_Confid  

[69 rows x 29 columns]

### Yangi column qo'shish

!! Natija(amal) mantiqsiz bo'lishi mumkin. Chunki o'zim o'ylab topdim

In [37]:
df['Activity'] = df['PhysicalActivity'] - df['SleepQuality']
df

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0          5034   63       0          1               0  15.848744        0   
1          5035   26       1          2               2  22.757042        0   
2          5036   57       0          2               1  18.395396        0   
3          5037   40       1          2               1  38.515278        0   
4          5038   61       0          0               3  19.283802        0   
...         ...  ...     ...        ...             ...        ...      ...   
2387       7421   43       1          0               2  29.059613        0   
2388       7422   18       1          0               1  20.740850        0   
2389       7423   54       0          3               2  37.079560        0   
2390       7424   46       1          0               2  23.444712        0   
2391       7425   26       1          0               0  28.123021        1   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFVC  \
0             0.894448     5.488696      8.701003  ...         4.941206   
1             5.897329     6.341014      5.153966  ...         1.702393   
2             6.739367     9.196237      6.840647  ...         5.022553   
3             1.404503     5.826532      4.253036  ...         2.300159   
4             4.604493     3.127048      9.625799  ...         3.067944   
...                ...          ...           ...  ...              ...   
2387          3.019854     6.119637      8.300960  ...         5.166032   
2388          5.805180     4.386992      7.731192  ...         5.509502   
2389          4.735169     8.214064      7.483521  ...         3.346877   
2390          9.672637     7.362861      6.717272  ...         1.713274   
2391          1.613138     7.412878      8.512253  ...         2.453284   

      Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
0            0                  0               1         0   
1            1                  0               0         1   
2            1                  1               1         0   
3            1                  0               1         1   
4            1                  1               1         0   
...        ...                ...             ...       ...   
2387         0                  1               0         0   
2388         0                  0               0         1   
2389         1                  0               1         1   
2390         0                  1               1         0   
2391         1                  1               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  Activity  
0                     0                1          0       Dr_Confid -7.806554  
1                     1                1          0       Dr_Confid  0.743363  
2                     1                1          0       Dr_Confid -0.101280  
3                     1                0          0       Dr_Confid -2.848533  
4                     0                1          0       Dr_Confid -5.021307  
...                 ...              ...        ...             ...       ...  
2387                  0                1          1       Dr_Confid -5.281106  
2388                  1                0          1       Dr_Confid -1.926012  
2389                  0                1          1       Dr_Confid -2.748352  
2390                  1                1          0       Dr_Confid  2.955365  
2391                  1                0          0       Dr_Confid -6.899115  

[2392 rows x 30 columns]

### Multiple Manipulations

In [38]:
physical_activity_until_5 = df[df['PhysicalActivity'] <= 5]
physical_activity_until_5_age = physical_activity_until_5.sort_values('Age', ascending=False)
physical_activity_until_5_age[['PatientID', 'Age', 'PhysicalActivity']]

PatientID  Age  PhysicalActivity
591        5625   79          2.106627
1251       6285   79          2.812983
1933       6967   79          3.261688
2176       7210   79          1.150078
368        5402   79          4.102267
...         ...  ...               ...
2377       7411    5          4.571440
1438       6472    5          4.372488
1579       6613    5          2.182973
703        5737    5          2.048262
1608       6642    5          4.060767

[1191 rows x 3 columns]